fn.go

```diff
@@ -20,34 +32,204 @@ type Function struct {
+
+		// Update only the status of the desired composite resource.
+		if cd.Resource.GetAPIVersion() == dxr.Resource.GetAPIVersion() && cd.Resource.GetKind() == dxr.Resource.GetKind() {
+			dst := make(map[string]any)
+			if err := dxr.Resource.GetValueInto("status", &dst); err != nil && !fieldpath.IsNotFound(err) {
+				response.Fatal(rsp, errors.Wrap(err, "cannot get desired composite status"))
+				return rsp, nil
+			}
+
+			src := make(map[string]any)
+			if err := cd.Resource.GetValueInto("status", &src); err != nil && !fieldpath.IsNotFound(err) {
+				response.Fatal(rsp, errors.Wrap(err, "cannot get desired composite status"))
+				return rsp, nil
+			}
+
+			for k, v := range src {
+				fmt.Println(k, v)
+				dst[k] = v
+			}
+
+			if err := fieldpath.Pave(dxr.Resource.Object).SetValue("status", dst); err != nil {
+				response.Fatal(rsp, errors.Wrap(err, "cannot set desired composite status"))
+				return rsp, nil
+			}
+
+			continue
+		}
+
+		// Set composite resource's connection details.
+		if cd.Resource.GetAPIVersion() == metaApiVersion {
+			switch obj.GetKind() {
+			case "CompositeConnectionDetails":
+				con, _ := cd.Resource.GetStringObject("data")
+				for k, v := range con {
+					d, _ := base64.StdEncoding.DecodeString(v) //nolint:errcheck // k8s returns secret values encoded
+					dxr.ConnectionDetails[k] = d
+				}
+			default:
+				response.Fatal(rsp, fmt.Errorf(errFmtInvalidMetaType, obj.GetKind()))
+				return rsp, nil
+			}
+
+			continue
+		}
+
+		// Set ready state.
+		if v, found := cd.Resource.GetAnnotations()[annotationKeyReady]; found {
+			if v != string(resource.ReadyTrue) && v != string(resource.ReadyUnspecified) && v != string(resource.ReadyFalse) {
+				response.Fatal(rsp, fmt.Errorf(fmt.Sprintf(errFmtInvalidFunction, errFmtInvalidReadyValue), v))
+				return rsp, nil
+			}
+
+			cd.Ready = fn.Ready(v)
+
+			// remove meta annotation
+			ann := cd.Resource.GetAnnotations()
+			delete(ann, annotationKeyReady)
+			cd.Resource.SetAnnotations(ann)
+		}
+
+		// Add resource to the desired composed resources map.
+		name, err := getCompositionResourceName(obj)
+		if err != nil {
+			response.Fatal(rsp, errors.Wrapf(err, "cannot get composition resource name of %s", obj.GetName()))
+			return rsp, nil
+		}
+
+		dcd[name] = cd
+	}
+
+	f.log.Debug("desired composite resource", "desiredComposite:", dxr)
+	f.log.Debug("constructed desired composed resources", "desiredComposed:", dcd)
+
+	if err := response.SetDesiredComposedResources(rsp, dcd); err != nil {
+		response.Fatal(rsp, errors.Wrap(err, "cannot desired dsd composed resources"))
+		return rsp, nil
+	}
+
+	if err := response.SetDesiredCompositeResource(rsp, dxr); err != nil {
+		response.Fatal(rsp, errors.Wrap(err, "cannot set desired composite resource"))
+		return rsp, nil
+	}
+
+	response.Normalf(rsp, "I was run with cd source %q", in.Source)
 
 	return rsp, nil
 }
+
+func convertToMap(req *fnv1beta1.RunFunctionRequest) (map[string]interface{}, error) {
```

Comment:
```suggestion
func convertToMap(req *fnv1beta1.RunFunctionRequest) (map[string]any, error) {
```
Also, other `interface{}` instances should be [changed](https://medium.com/@csaba.ujvari/go-1-18-any-instead-of-interface-ce15dd7bea5d) to `any`.